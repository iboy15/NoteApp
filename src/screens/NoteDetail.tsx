import { AntDesign } from '@expo/vector-icons'
import { useRoute } from '@react-navigation/native'
import { useState } from 'react'
import { StyleSheet, TextInput, TouchableOpacity, View } from 'react-native'
import { NoteEdit } from '../../types'
import { Text } from '../components/Themed'
import Colors from '../constants/Colors'
import useColorScheme from '../hooks/useColorScheme'
import {useMain} from '../hooks/useMain'

export

const NoteDetail = () => {
  const props: NoteEdit = useRoute().params as NoteEdit;
  const colorScheme = useColorScheme()
  const [editable, setEditable] = useState(false)
  const [title, setTitle] = useState<string>(props.title || '')
  const [body, setBody] = useState<string>(props.body || '')
  const { onEdit, onDelete } = useMain()

  return (
    <View style={{ flex: 1, backgroundColor: props.color, padding: 20 }}>
      <TextInput
        editable={editable}
        style={[
          styles.title,
          {
            color: Colors[colorScheme].text,
            borderColor: editable ? Colors[colorScheme].text : props.color,
          },
        ]}
        value={title}
        onChangeText={setTitle}
      />

      <TextInput
        numberOfLines={15}
        scrollEnabled
        textAlignVertical='top'
        multiline
        editable={editable}
        style={[
          styles.body,
          {
            color: Colors[colorScheme].text,
            borderColor: editable ? Colors[colorScheme].text : props.color,
          },
        ]}
        value={body}
        onChangeText={setBody}
      />
      <View style={styles.saveContainer}>
        {editable ? (
          <TouchableOpacity
            onPress={() => onEdit({title, body, index: props.index})}
            style={[
              styles.saveBtn,
              {
                backgroundColor: Colors[colorScheme].background,
              },
            ]}
          >
            <Text style={{ fontWeight: 'bold' }}>Save</Text>
          </TouchableOpacity>
        ) : (
          <View />
        )}
        {props.updated_at &&
        <Text style={{marginTop:20}}>{props.updated_at}</Text>
        }
        <View
          style={[
            styles.rightMenuContainer,
            {
              backgroundColor: Colors[colorScheme].background,
            },
          ]}
        >
          <TouchableOpacity
            onPress={() => setEditable(!editable)}
            style={{ marginRight: 20 }}
          >
            <AntDesign
              name='edit'
              color={editable ? Colors.light.tint : Colors[colorScheme].text}
              size={24}
            />
            <Text
              style={[
                styles.editText,
                {
                  color: editable
                    ? Colors.light.tint
                    : Colors[colorScheme].text,
                },
              ]}
            >
              Edit
            </Text>
          </TouchableOpacity>
          <TouchableOpacity
            onPress={() => onDelete(props.index)}
            style={{ alignItems: 'center' }}
          >
            <AntDesign
              name='delete'
              color={Colors[colorScheme].text}
              size={24}
            />
            <Text style={{ marginTop: 5, fontSize: 12 }}>Delete</Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  )
}

export default NoteDetail

const styles = StyleSheet.create({
  title: {
    fontWeight: 'bold',
    fontSize: 20,
    paddingVertical: 10,
    borderWidth: 1,
    borderRadius: 10,
  },
  body: {
    paddingVertical: 15,
    borderWidth: 1,
    borderRadius: 10,
    marginTop: 20,
    paddingHorizontal: 5,
  },
  saveContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  saveBtn: {
    paddingVertical: 20,
    paddingHorizontal: 30,
    borderRadius: 10,
    top: 10,
  },
  rightMenuContainer: {
    marginTop: 30,
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 10,
    paddingVertical: 10,
    paddingHorizontal: 15,
    borderRadius: 10,
  },
  editText: {
    marginTop: 5,
    fontSize: 12,
  },
})
