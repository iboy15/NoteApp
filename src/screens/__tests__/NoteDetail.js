import * as React from 'react';
import renderer from 'react-test-renderer';
import NoteDetail from '../NoteDetail'
import MockedNavigator from '../../navigation/MockedNavigator'

it(`renders correctly`, async () => {
  const tree = await renderer.create(<MockedNavigator  component={NoteDetail} />).toJSON();

  expect(tree).toMatchSnapshot();
});
