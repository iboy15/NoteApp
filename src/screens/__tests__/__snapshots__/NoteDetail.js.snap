// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="MockedScreen"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": undefined,
              "flex": 1,
              "padding": 20,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={false}
            onChangeText={[Function]}
            rejectResponderTermination={true}
            style={
              Array [
                Object {
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "paddingVertical": 10,
                },
                Object {
                  "borderColor": undefined,
                  "color": "#000",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
          <TextInput
            allowFontScaling={true}
            editable={false}
            multiline={true}
            numberOfLines={15}
            onChangeText={[Function]}
            rejectResponderTermination={true}
            scrollEnabled={true}
            style={
              Array [
                Object {
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "marginTop": 20,
                  "paddingHorizontal": 5,
                  "paddingVertical": 15,
                },
                Object {
                  "borderColor": undefined,
                  "color": "#000",
                },
              ]
            }
            textAlignVertical="top"
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 10,
                    "flexDirection": "row",
                    "marginBottom": 10,
                    "marginTop": 30,
                    "paddingHorizontal": 15,
                    "paddingVertical": 10,
                  },
                  Object {
                    "backgroundColor": "#fff",
                  },
                ]
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "marginRight": 20,
                    "opacity": 1,
                  }
                }
              >
                <Text />
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000",
                      },
                      Array [
                        Object {
                          "fontSize": 12,
                          "marginTop": 5,
                        },
                        Object {
                          "color": "#000",
                        },
                      ],
                    ]
                  }
                >
                  Edit
                </Text>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "opacity": 1,
                  }
                }
              >
                <Text />
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000",
                      },
                      Object {
                        "fontSize": 12,
                        "marginTop": 5,
                      },
                    ]
                  }
                >
                  Delete
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
